jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/JdbcAdapter.java

```diff
@@ -1,273 +1,276 @@
+    }
+
+    private static void registerAdapterForSqlDialect(final VirtualSchemaAdapter adapter, final String dialectName) {
+        AdapterRegistry.getInstance().registerAdapter(dialectName, adapter);
+    }
+
+    @Override
+    public CreateVirtualSchemaResponse createVirtualSchema(final ExaMetadata exasolMetadata,
+            final CreateVirtualSchemaRequest request) throws AdapterException {
+        logCreateVirtualSchemaRequestReceived(request);
+        final SchemaMetadataInfo schemaMetadataInfo = request.getSchemaMetadataInfo();
+        final AdapterProperties properties = getPropertiesFromRequest(request);
+        JdbcAdapterProperties.checkPropertyConsistency(schemaMetadataInfo.getProperties());
         try {
-            final AdapterRequest request = new RequestJsonParser().parseRequest(input);
-            final SchemaMetadataInfo schemaMetadata = request.getSchemaMetadataInfo();
-            configureLogOutput(schemaMetadata);
-            logger.fine(() -> "Adapter request:\n" + input);
+            final SchemaMetadata remoteMeta = readMetadata(properties, exasolMetadata);
+            return CreateVirtualSchemaResponse.builder().schemaMetadata(remoteMeta).build();
+        } catch (final SQLException exception) {
+            throw new AdapterException("Unable create Virtual Schema \"" + request.getVirtualSchemaName() + "\".",
+                    exception);
+        }
+    }
+
+    protected void logCreateVirtualSchemaRequestReceived(final CreateVirtualSchemaRequest request) {
+        LOGGER.fine(() -> "Received request to create Virutal Schema \"" + request.getVirtualSchemaName() + "\".");
+    }
 
-            switch (request.getType()) {
-            case CREATE_VIRTUAL_SCHEMA:
-                result = handleCreateVirtualSchema((CreateVirtualSchemaRequest) request, meta);
-                break;
-            case DROP_VIRTUAL_SCHEMA:
-                result = handleDropVirtualSchema((DropVirtualSchemaRequest) request);
-                break;
-            case REFRESH:
-                result = handleRefresh((RefreshRequest) request, meta);
-                break;
-            case SET_PROPERTIES:
-                result = handleSetProperty((SetPropertiesRequest) request, meta);
-                break;
-            case GET_CAPABILITIES:
-                result = handleGetCapabilities((GetCapabilitiesRequest) request);
-                break;
-            case PUSHDOWN:
-                result = handlePushdownRequest((PushdownRequest) request, meta);
-                break;
-            default:
-                throw new RuntimeException("Request Type not supported: " + request.getType());
+    private AdapterProperties getPropertiesFromRequest(final AdapterRequest request) {
+        return new AdapterProperties(request.getSchemaMetadataInfo().getProperties());
+    }
+
+    private SchemaMetadata readMetadata(final AdapterProperties properties, final ExaMetadata exasolMetadata)
```

Comment:
There are still some helper methods in the JbdcAdapter class. I would like the JdbcAdapter just to implement the VirtualSchemaAdapter. Don't know if that is feasible.

Reply:
A lot actually. But I think we tackle those in the next refactoring round.